physionet-django/events/views.py

```diff
@@ -235,3 +235,58 @@ def event_detail(request, event_slug):
          'is_waitlisted': is_waitlisted,
          'event_datasets': event_datasets,
          })
+
+
+@login_required
+def manage_co_hosts(request):
+    """
+    Manage co-hosts of an event
+    """
+    user = request.user
+
+    if request.method == 'POST' and request.is_ajax():
+        participant_id = request.POST.get('participant_id')
+
+        event_slug = request.POST.get('event_slug')
+        event = get_object_or_404(Event, slug=event_slug)
+
+        if not event.host == user:
+            return JsonResponse({'error': 'You are not the host of this event'}, status=403)
+
+        if event.end_date < datetime.now().date():
+            return JsonResponse({'error': 'You cannot manage co-hosts of an event that has ended'}, status=403)
+
+        if not event.participants.filter(id=participant_id).exists():
+            return JsonResponse({'error': 'User is not a participant of this event'}, status=403)
+
+        participant = event.participants.get(id=participant_id)
+
+        if 'Remove cohost' in request.POST.get('submit'):
+            if not participant.is_cohost:
+                return JsonResponse({'error': 'User is not a cohost of this event'}, status=403)
+            participant.is_cohost = False
+            participant.save()
+            # notify the cohost that their cohost permission has been removed
+            notification.notify_event_cohost_cohost_status_change(request=request, cohost=participant.user,
+                                                                  event=event, status='Remove cohost')
+            # notify the host that they have removed a cohost
+            notification.notify_event_host_cohost_status_change(request=request, cohost=participant.user, event=event,
+                                                                status='Remove cohost')
+
+            return JsonResponse({'success': 'Cohost removed successfully'})
+        elif 'Make cohost' in request.POST.get('submit'):
+            if participant.is_cohost:
+                return JsonResponse({'error': 'User is already a cohost of this event'}, status=403)
+            participant.is_cohost = True
+            participant.save()
+            # notify the cohost that they have been added as a cohost for the event
+            notification.notify_event_cohost_cohost_status_change(request=request, cohost=participant.user,
+                                                                  event=event, status='Make cohost')
+            # notify the host that they have added a cohost for the event
```

Comment:
I don't see a point in adding comments that just explain code that should be self-documenting. I'd remove them.

Reply:
Removed the comments.